fl4health/model_bases/masked_layers/masked_normalization_layers.py

```diff
@@ -0,0 +1,297 @@
+
+        # Decide whether the mini-batch stats should be used for normalization rather than the buffers.
+        # Mini-batch stats are used in training mode, and in eval mode when buffers are None.
+        if self.training:
+            bn_training = True
+        else:
+            bn_training = (self.running_mean is None) and (self.running_var is None)
+
+        # Buffers are only updated if they are to be tracked and we are in training mode. Thus they only need to be
+        # passed when the update should occur (i.e. in training mode when they are tracked), or when buffer stats are
+        # used for normalization (i.e. in eval mode when buffers are not None).
+        if self.affine:
+            assert (self.weight is not None) and (self.bias is not None)
+            weight_prob_scores = torch.sigmoid(self.weight_scores)
+            weight_mask = bernoulli_sample(weight_prob_scores)
+            masked_weight = weight_mask * self.weight
+
+            bias_prob_scores = torch.sigmoid(self.bias_scores)
+            bias_mask = bernoulli_sample(bias_prob_scores)
+            masked_bias = bias_mask * self.bias
+        else:
+            masked_weight = None
+            masked_bias = None
+
+        return F.batch_norm(
+            input,
+            # If buffers are not to be tracked, ensure that they won't be updated
+            self.running_mean if not self.training or self.track_running_stats else None,
+            self.running_var if not self.training or self.track_running_stats else None,
+            masked_weight,
+            masked_bias,
+            bn_training,
+            exponential_average_factor,
+            self.eps,
+        )
+
+    @classmethod
+    def from_pretrained(cls, batch_norm_module: _BatchNorm) -> "_MaskedBatchNorm":
+        masked_batch_norm_module = cls(
+            num_features=batch_norm_module.num_features,
+            eps=batch_norm_module.eps,
+            momentum=batch_norm_module.momentum,
+            affine=batch_norm_module.affine,
+            track_running_stats=batch_norm_module.track_running_stats,
+        )
+        if batch_norm_module.affine:
+            assert (batch_norm_module.weight is not None) and (batch_norm_module.bias is not None)
+            masked_batch_norm_module.weight = Parameter(batch_norm_module.weight.clone().detach(), requires_grad=False)
+            masked_batch_norm_module.weight_scores = Parameter(
+                torch.randn_like(batch_norm_module.weight), requires_grad=True
+            )
+            masked_batch_norm_module.bias = Parameter(batch_norm_module.bias.clone().detach(), requires_grad=False)
+            masked_batch_norm_module.bias_scores = Parameter(
+                torch.randn_like(batch_norm_module.weight), requires_grad=True
+            )
+        return masked_batch_norm_module
+
+
+class MaskedBatchNorm1d(_MaskedBatchNorm):
+    """
+    Applies (masked) Batch Normalization over a 2D or 3D input.
```

Comment:
Maybe we can mention that the 2D is batch x input input and 3D is batch x input x channels?